include/zephyr/bluetooth/classic/classic.h

```diff
@@ -196,13 +196,79 @@ int bt_br_set_discoverable(bool enable, bool limited);
  * Allows make local controller to be connectable. It means the controller
  * start listen to devices requests on PAGE SCAN channel. If disabled also
  * resets discoverability if was set.
+*
+* @param enable Value allowing/disallowing controller to be connectable.
+*
+* @return Negative if fail set to requested state or requested state has been
+*         already set. Zero if done successfully.
+*/
+nt bt_br_set_connectable(bool enable);
+/**
+ * @brief Set controller page scan activity.
+ *
+ * Page Scan is only performed when Page_Scan is enabled.
+ *
+ * @param interval Page scan interval in 0.625 ms units
+ *        Range: 0x0012 to 0x1000; only even values are valid.
+ * @param window Page scan window in 0.625 ms units
+ *        Range: 0x0011 to 0x1000.
+ *
+ * @return Negative if fail set to requested state or requested state has been
+ *         already set. Zero if done successfully.
+ */
+int bt_br_write_page_scan_activity(uint16_t interval, uint16_t window);
+
+/**
+ * @brief Set controller inquiry scan activity.
+ *
+ * Inquiry Scan is only performed when Inquiry_Scan is enabled.
+ *
+ * @param interval Inquiry scan interval in 0.625 ms units
+ *        Range: 0x0012 to 0x1000; only even values are valid.
+ * @param window Inquiry scan window in 0.625 ms units
+ *        Range: 0x0011 to 0x1000.
+ *
+ * @return Negative if fail set to requested state or requested state has been
+ *         already set. Zero if done successfully.
+ */
+int bt_br_write_inquiry_scan_activity(uint16_t interval, uint16_t window);
+
+/**
+ * @brief Set the inquiry Scan Type configuration parameter of the local
+ *        BR/EDR Controller.
+ *
+ * @param type Inquiry scan type.
+ * 		   0x00: Standard scan (default)
+ * 		   0x01: Interlaced scan
+ *
+ * @return Negative if fail set to requested state or requested state has been
+ *         already set. Zero if done successfully.
+ */
+int bt_br_write_inquiry_scan_type(uint8_t type);
+
+/**
+ * @brief Set the page Scan Type configuration parameter of the local
+ *        BR/EDR Controller.
+ *
+ * @param type Page scan type.
+ *        0x00: Standard scan (default)
+ *        0x01: Interlaced scan
+ *
+ * @return Negative if fail set to requested state or requested state has been
+ *         already set. Zero if done successfully.
+ */
+int bt_br_write_page_scan_type(uint8_t type);
```

Comment:
IMO these are too low level for the host stack API. They're essentially just 1:1 mapping to HCI. Please consider how we can abstract these in some nicer ways and if there are existing (possibly generic to both LE and BR/EDR) APIs that could be used instead.

Reply:
We could potentially have a single API to set all three parameters, with some config struct as input, and then have helper macros to construct those standard "Scenario" combinations from that table. That would retain the high level friendliness, but still allow someone to go and do some custom stuff if they really need it.